### DISCUSSION TITLE
Allow rendering HTML in the exercise notes (allow to include gifs, image and videos)

### DISCUSSION THREAD

**Discussion by felipegrin** [LiamMorrow/LiftLog#823] [Ideas]
It would be nice to allow entering HTML in the exercise notes. This way it will be possible to embed a youtube video for the exercise, a GIF or a static image of the exercise (hosted in another website).

This looks to me as a very low effort and a really useful feature. 

Right now, if I try entering HTML to the exercise notes, it escapes the HTML instead of rendering the content. See below the screenshots.

**HTML in the exercise notes:**

![edit](https://github.com/user-attachments/assets/bcbd103b-4774-4e7e-8d80-86c993ce9142)

**Result:**

![result](https://github.com/user-attachments/assets/0199db08-6274-4a70-bb9b-d8d1fea4468c)

**Expected (mockup):**

![expected](https://github.com/user-attachments/assets/2dbba3be-e35b-4c5c-9c41-ea1ab6a95762)

**Comment by LiamMorrow** [comment]
Hi @felipegrin ! I agree, this could be a very powerful feature! I fear a little about using HTML directly (there are security concerns with shared plans/feed with script injection we would have to deal with). Also it's a little cumbersome, especially on a phone.  Are you aware of the External link field on the exercise? This allows you to add a link which can be opened directly from the workout page:
<img width="358" alt="image" src="https://github.com/user-attachments/assets/bcc83d42-d482-43e1-a2a1-8c1160cfc5c5" />


It's not directly inline, but you could add a link to an image in that field

**Comment by felipegrin** [comment]
Thank you for your reply! I'm currently doing that, the external link helps a lot. However, it would be better to have the content directly on the exercise page so I wouldn't need to keep going back and forth :)

**Comment by LiamMorrow** [comment]
Extremely fair. I'll have a think about it for sure.  I'm not entirely against the idea of using something like markdown, however it has its own usability issues (and people who don't know about it could be confused as to why some things they type cause different effects).  A wysiwyg editor would be ideal if we do go down this "rich notes" path.

**Comment by yogeshwar-b** [comment]
@LiamMorrow Sorry for Hijacking issue here, But this seemed like a relevant thing. Do you have a plan of adding workout images in future in the LiftLog app ? 

If you are worried about licensing, even AI Gen images/gifs will work.

**Comment by LiamMorrow** [comment]
All good! At this point all work is being put into the react native rewrite. After that I'll be looking into setting up a roadmap with priorities and what's planned. This isn't ruled out though, it just needs to be prioritised

**Comment by yogeshwar-b** [comment]
Will you move React Native port to a new repo? I would like to contribute.
